source/validate.cpp

```diff
@@ -286,6 +291,12 @@ spv_result_t spvValidateIDs(const spv_instruction_t* pInsts,
 }
 
 namespace {
+// Shame
```

Comment:
Hmm. it looks like it complains if the error variable is defined outside of the if statement but does not if you define it in the statement.

```
//warning
spv_result_t ret;
if(ret = exp) return ret;

// ok
if(spv_result_t ret = exp) return ret;
```
I will update the macro. Thanks :)